File: java/estruturas-repetitivas/Quadrante.java

import java.util.Locale;
import java.util.Scanner;

public class Quadrante {
    public static void main(String[] args) {
        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        double x, y;

        System.out.println("Digite os valores das coordenadas X e Y: ");
        x = sc.nextDouble();
        y = sc.nextDouble();

        do {
            if (x > 0 && y > 0) {
                System.out.println("QUADRANTE Q1");
            }

            if (x < 0 && y > 0) {
                System.out.println("QUADRANTE Q2");
            }

            if (x < 0 && y < 0) {
                System.out.println("QUADRANTE Q3");
            }

            if (x > 0 && y < 0) {
                System.out.println("QUADRANTE Q4");
            }

            System.out.println("Digite os valores das coordenadas X e Y: ");
            x = sc.nextDouble();
            y = sc.nextDouble();
        } while (x != 0 && y != 0);
    }
}

File: js/estruturas-repetitivas/quadrante.js
const readline = require('readline-sync');

console.log("Digite os valores das coordenadas X e Y:");
let X = Number(readline.question());
let Y = Number(readline.question());

do {
    if (X > 0 && Y > 0) {
        console.log("QUADRANTE Q1");
    }

    if (X < 0 && Y > 0) {
        console.log("QUADRANTE Q2");
    }

    if (X < 0 && Y < 0) {
        console.log("QUADRANTE Q3");
    }

    if (X > 0 && Y < 0) {
        console.log("QUADRANTE Q4");
    }
    console.log("Digite os valores das coordenadas X e Y:");
    X = Number(readline.question());
    Y = Number(readline.question());
} while (X !== 0 && Y !== 0);